### PR TITLE
try append noreply arg?

### PR DESCRIPTION
**Pull request by ArtemIsmagilov**
- Done formating
### TL;DR - What are you trying to accomplish?

- change signature and body command delete with noreply arg, type bool 
### Details - How are you making this change?  What are the effects of this change?
I would like to suggest that you explicitly specify the noreply parameter because it is a parameter for the command, not a separate command. It also makes sense because otherwise you need a 2-copy implementation for all `Retrieval` and `Delete` commands that can include this flag.

The amount of code otherwise grows progressively, along with fixes.

But what I don't like about my solution? Two separate functions with different flags are easier to read, lighter, stricter and more stable.

There is a compromise between the strict implementation of all separate functions with different parameters, which ensures reliability, and on the other hand, an API similar to memcached and a reduction in code.

What do you think about this?

### CODE REVIEW

**Comment by ArtemIsmagilov** [comment]
I think that in this matter, probably the main thing is that it works flawlessly and does not lead to dependencies and pasta complications

**Review by nickamorim — COMMENTED** [review]
Considering it is considered an [anti-pattern using `noreply` with the ASCII protocol](https://github.com/memcached/memcached/wiki/Commands#no-reply), my preference is leaving the `noreply` variant as a separate method.

**Comment by ArtemIsmagilov** [comment]
Thanks for sharing this article. It really shouldn't be the way I did it. For all ASCII commands we need to implement 26 public functions.

Thanks again for sharing the article. Binary is of course no longer used or supported
